File: src/DexaWeb.Server/Models/Dexa/AssetType.cs
namespace DexaWeb.Server.Models.Dexa;

public class AssetType
{
    public int? Id { get; set; }
    public string? UserFriendlyName { get; set; }
    public string? Guid { get; set; }
    public bool Fake { get; set; }
    public string? Icon { get; set; }
    public string? Parameter { get; set; }
    public string? DotnetClassName { get; set; }
    public string? Description { get; set; }
}

File: src/Twms2.Dexa/Models/AssetType.fs
namespace DEX.Core.Database.ORM

open System

type AssetType() =
    member val Id: Nullable<int> = Nullable<int>() with get, set
    member val Guid: string = null with get, set
    member val Fake: bool = false with get, set
    member val UserFriendlyName: string = null with get, set
    member val Parameter: string = null with get, set
    member val Icon: string = null with get, set
    member val DotnetClassName: string = null with get, set
    member val Description: string = null with get, set

    new(id: Nullable<int>,
        guid: string,
        userFriendlyName: string,
        parameter: string,
        icon: string,
        dotnetClassName: string) as this =
        AssetType()
        then
            this.Id <- id
            this.Guid <- guid
            this.UserFriendlyName <- userFriendlyName
            this.Parameter <- parameter
            this.Icon <- icon
            this.DotnetClassName <- dotnetClassName
